src/Components/ArticleCard/ArticleCard.jsx

```diff
@@ -0,0 +1,47 @@
+import { React, useState } from "react";
+import { Link } from "react-router-dom";
+import { Icon } from "@iconify-icon/react";
+import styles from "./ArticleCard.module.scss";
+import TagItem from "../TagItem/TagItem";
+
+const ArticleCard = ({ data }) => {
+  const [bookmarked, setBookmark] = useState(data?.bookmarked);
+  const handleBookmark = () => {
+    /* Make requests to backend to update bookmark */
+    setBookmark(!bookmarked);
```

Comment:
It's better (best practice) to use callback version of updating the state when you want to update the state based on previous value.

Refer to this:
https://stackoverflow.com/a/55342704